src/html/to_html_tree.ml

```diff
@@ -684,15 +684,14 @@ struct
         let html = Html.dt ~a html in
         let acc = html::acc in
 
-        begin match maybe_docs with
-        | [] ->
-          consume_leaf_items_until_one_is_documented items acc
-        | docs ->
-          let docs = Documentation.to_html docs in
-          let docs = (docs :> (Html_types.dd_content Html.elt) list) in
-          let docs = Html.dd docs in
-          List.rev (docs::acc), items
-        end
+        let docs = Documentation.to_html maybe_docs in
```

Comment:
Does this change make it so each `<dt>` tag is followed by `<dd>`, whether there are docs or not?

If so, I don't think we should do it.

Sometimes, there are multiple signature items, that are summarized by one doc comment:

```ocaml
type yes
type no
(** Phantom types for some purpose. *)
```

In this case, I think we'd like

```html
<dt><!-- type yes --></dt>
<dt><!-- type no --></dt>
<dd><!-- docs for both --></dd>
```

This helps with styling as well, as we can use CSS to avoid vertical space between the two types, making them look visually grouped.

Here's a live example: http://aantron.github.io/lambda-soup/#TYPEelement

WhatWG suggests this is acceptable as well: https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element, see "In the following example, one definition is linked to two terms."

Reply:
Yes, that's exactly what it's doing to compensate for the `missing <dd>` warnings.

This is starting to point towards capturing the output and searching for the warnings we consider intolerable.

Reply:
@ostera, I just installed tidy locally. It looks like with the right `--mute` options, the exit code is still 1, but STDERR is empty. So we don't have to parse the output, just check that it's empty. If non-empty, we show it to the user and fail the test.

Reply:
Updated.